CommuniTEAM/CommuniTEA
# [TEA-84] Fix Faulty CORS Setup

### In this PR, I'm adding/changing the following:

- Moved CORS middleware setup to apply to handlers individually, rather than the API application as a whole
- Tightened CORS request origin security to differentiate between production and development environments
- Allowed PATCH requests to accommodate new endpoints
- Upgraded out of date Chi middleware package to v5
- **Updated Go to 1.22!**

### Please do the following to test my code:

1. Delete and rebuild docker containers!
2. Make sure the API still works as expected
3. Merge this branch into another working branch to test frontend-backend communication with Axios
4. You should not encounter any CORS errors

### Notes

**Important:** Since the version of Go has changed, it is imperative that you update your local version of Go and golangci-lint to match.

Go can be updated via VS Code as seen in the screenshot below:
![Updating Go](https://github.com/CommuniTEAM/CommuniTEA/assets/31549337/c8d61ee9-fb50-443c-a4d8-b0e74bc53e85)


If on Mac or Linux, use brew to update golangci-lint to version 1.56.2. If using Windows, run this command in Git Bash: 
```
curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin 

golangci-lint --version
```

[Related Jira Ticket](https://communiteam.atlassian.net/browse/TEA-84)
